cmd/in_memory/channel_dispatcher/main.go

```diff
@@ -17,166 +17,16 @@ limitations under the License.
-			opt,
-			inMemoryDispatcher,
-			inMemoryChannelInformer,
-		),
-	}
-	// This line asserts at compile time that the length of controllers is equal to numControllers.
-	// It is based on https://go101.org/article/tips.html#assert-at-compile-time, which notes that
-	// var _ [N-M]int
-	// asserts at compile time that N >= M, which we can use to establish equality of N and M:
-	// (N >= M) && (M >= N) => (N == M)
-	var _ [numControllers - len(controllers)][len(controllers) - numControllers]int
-
-	// Watch the logging config map and dynamically update logging levels.
-	opt.ConfigMapWatcher.Watch(logconfig.ConfigMapName(), logging.UpdateLevelFromConfigMap(logger, atomicLevel, logconfig.Controller))
-	// TODO: Watch the observability config map and dynamically update metrics exporter.
-	//opt.ConfigMapWatcher.Watch(metrics.ObservabilityConfigName, metrics.UpdateExporterFromConfigMap(component, logger))
-
-	// Setup zipkin tracing.
-	if err = tracing.SetupDynamicZipkinPublishing(logger, opt.ConfigMapWatcher, "imc-dispatcher"); err != nil {
-		logger.Fatalw("Error setting up Zipkin publishing", zap.Error(err))
-	}
-
-	if err := opt.ConfigMapWatcher.Start(stopCh); err != nil {
-		logger.Fatalw("failed to start configuration manager", zap.Error(err))
-	}
-
-	// Start all of the informers and wait for them to sync.
-	logger.Info("Starting informers.")
-	if err := kncontroller.StartInformers(
-		stopCh,
-		// Messaging
-		inMemoryChannelInformer.Informer(),
-	); err != nil {
-		logger.Fatalf("Failed to start informers: %v", err)
-	}
-
-	go inMemoryDispatcher.Start(stopCh)
-
-	logger.Info("Starting controllers.")
-	kncontroller.StartAll(stopCh, controllers[:]...)
-
-	inMemoryDispatcher.Stop()
-}
-
-func setupLogger() (*zap.SugaredLogger, zap.AtomicLevel) {
-	// Set up our logger.
-	loggingConfigMap := getLoggingConfigOrDie()
-	loggingConfig, err := logging.NewConfigFromMap(loggingConfigMap)
-	if err != nil {
-		log.Fatalf("Error parsing logging configuration: %v", err)
-	}
-	return logging.NewLoggerFromConfig(loggingConfig, logconfig.Controller)
-}
-
-func getLoggingConfigOrDie() map[string]string {
-	if hardcodedLoggingConfig != nil && *hardcodedLoggingConfig {
-		return map[string]string{
-			"loglevel.controller": "info",
-			"zap-logger-config": `
-				{
-					"level": "info",
-					"development": false,
-					"outputPaths": ["stdout"],
-					"errorOutputPaths": ["stderr"],
-					"encoding": "json",
-					"encoderConfig": {
-					"timeKey": "ts",
-					"levelKey": "level",
-					"nameKey": "logger",
-					"callerKey": "caller",
-					"messageKey": "msg",
-					"stacktraceKey": "stacktrace",
-					"lineEnding": "",
-					"levelEncoder": "",
-					"timeEncoder": "iso8601",
-					"durationEncoder": "",
-					"callerEncoder": ""
-				}`,
-		}
-	} else {
-		cm, err := configmap.Load("/etc/config-logging")
-		if err != nil {
-			log.Fatalf("Error loading logging configuration: %v", err)
-		}
-		return cm
-	}
+	sharedmain.Main("inmemorychannel-dispatcher",
```

Comment:
We are losing the zipkin tracing with this change, right?

Reply:
good catch.